protobuf/lib/src/protobuf/json.dart

```diff
@@ -38,23 +39,26 @@ Map<String, dynamic> _writeToJsonMap(_FieldSet fs) {
         // Encode 'bytes' as a base64-encoded string.
         return base64Encode(fieldValue as List<int>);
       case PbFieldType._ENUM_BIT:
-        return fieldValue.value; // assume |value| < 2^52
+        ProtobufEnum enum_ = fieldValue;
+        return enum_.value; // assume |value| < 2^52
```

Comment:
@mkustermann we use the `dynamic` value (`fieldValue`) for one field access here. Do you think think the cast is still beneficial? I wonder if not casting here will be faster.

We have similar cases above.

Reply:
Here are the trade-offs for VM's AOT:

When using `dynamic` for `fieldValue.value`:
  * startup: we have to pay per-call site per-runtime-class overhead (go to C++, lookup function in dictionary, ...)
  * code size: all classes that have `value` getter/field will retain it (cannot tree shake) and possibly get an extra `dyn:get:value` function injected (that boxes the unboxed `int64` into a tagged pointer)
  * if many different receiver classes flow into the call site we will have to look up the target in a per-callsite hashmap (megamorphic cache)

When using `as ProtobufEnum`:
  * we perform a class-id range check - which is - especially in AOT - very cheap
  * we do a vtable like call and may get a unboxed `int64` (and therefore avoid boxing)